tools/cli/air/src/trace/run/risc0.rs

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use super::runner::AirRunner;
+
+use air_interpreter_interface::InterpreterOutcome;
+use air_interpreter_interface::RunParameters;
+use avm_interface::into_raw_result;
+use avm_interface::raw_outcome::RawAVMOutcome;
+use fluence_keypair::KeyPair;
+use zk_aquavm_interface::AquaVMProvingParameters;
+
+use risc0_zkvm::default_prover;
+use risc0_zkvm::ExecutorEnv;
+use zk_aquavm_methods::ZK_AQUAVM_ELF;
+use zk_aquavm_methods::ZK_AQUAVM_ID;
+
+use std::time::Instant;
+
+pub struct Risc0Runner {}
+
+impl Risc0Runner {
+    pub(crate) fn new() -> Self {
+        Self {}
+    }
+}
+
+impl AirRunner for Risc0Runner {
+    fn call_tracing(
+        &mut self,
+        air: String,
+        prev_data: Vec<u8>,
+        current_data: Vec<u8>,
+        init_peer_id: String,
+        timestamp: u64,
+        ttl: u32,
+        current_peer_id: String,
+        call_results: avm_interface::CallResults,
+        _tracing_params: String,
+        _tracing_output_mode: u8,
+        keypair: &KeyPair,
+        particle_id: String,
+    ) -> anyhow::Result<RawAVMOutcome> {
+        let key_format = keypair.key_format().into();
+        let secret_key_bytes = keypair.secret().expect("Failed to get secret key");
+
+        let run_params = RunParameters {
+            init_peer_id,
+            current_peer_id,
+            timestamp,
+            ttl,
+            key_format,
+            secret_key_bytes,
+            particle_id,
+        };
+
+        let call_results = into_raw_result(call_results);
+        let call_results = serde_json::to_vec(&call_results)?;
+
+        let arguments = AquaVMProvingParameters {
+            air,
+            prev_data,
+            current_data,
+            run_params,
+            call_results,
+        };
+
+        execute_on_risc0(arguments)
+    }
+}
+
+fn execute_on_risc0(arguments: AquaVMProvingParameters) -> anyhow::Result<RawAVMOutcome> {
+    use risc0_zkvm::serde::from_slice;
+
+    let env = ExecutorEnv::builder().write(&arguments)?.build()?;
+
+    let prover = default_prover();
+
+    eprintln!("default prover created, proving is started");
```

Comment:
Consider using a logger, e.g. `tracing::info!` (though it may give less readable output).

Reply:
These logs are intended b/c it tells that the long-running process has started. If you mean that every single output should be printed with `tracing`, in other parts of the CLI `eprintln` is using, so it's better to keep it consistent.